flink-runtime/src/main/java/org/apache/flink/runtime/metrics/util/MetricUtils.java

```diff
@@ -214,6 +219,29 @@ static void instantiateNonHeapMemoryMetrics(final MetricGroup metricGroup) {
 		instantiateMemoryUsageMetrics(metricGroup, () -> ManagementFactory.getMemoryMXBean().getNonHeapMemoryUsage());
 	}
 
+	@VisibleForTesting
+	static void instantiateMetaspaceMemoryMetrics(final MetricGroup parentMetricGroup) {
+		final List<MemoryPoolMXBean> memoryPoolMXBeans = ManagementFactory.getMemoryPoolMXBeans()
+			.stream()
+			.filter(bean -> "Metaspace".equals(bean.getName()))
+			.collect(Collectors.toList());
+
+		if (memoryPoolMXBeans.isEmpty()) {
+			LOG.warn("No memory pool named 'Metaspace' is present. The '{}' metric group is not going to be instantiated.", METRIC_GROUP_METASPACE_NAME);
```

Comment:
```suggestion
			LOG.info("Metaspace metrics will not be exposed because no pool named 'Metaspace' could be found. This could be due to the used JVM.");
```
Users typically don't like warning they can't do anything about, and the message is relatively cryptic ("what is a metric group? what is the consequence of it not being instantiated?"

Reply:
Thanks for clarification. Makes sense!